beacon-chain/blockchain/testing/mock.go

```diff
@@ -98,6 +99,9 @@ func (msn *MockStateNotifier) ReceivedEvents() []*feed.Event {
 
 // StateFeed returns a state feed.
 func (msn *MockStateNotifier) StateFeed() *event.Feed {
+	msn.feedLock.Lock()
+	defer msn.feedLock.Unlock()
```

Comment:
Added this to prevent potential race condition on accessing the state feed mock in test.